Commit a18e51ef4e4b "[net] Update API for VF vlan protocol 802.1ad support"
(upstream 79aab093a0b5) introduced new version of ndo_set_vf_vlan callback,
which adds VLAN protocol argument. qede, however, supports only 802.1q, so it
does not benefits from this change.
Index: src/drivers/net/ethernet/qlogic/qede/qede_main.c
===================================================================
--- src.orig/drivers/net/ethernet/qlogic/qede/qede_main.c	2017-04-19 17:55:46.080025847 +0200
+++ src/drivers/net/ethernet/qlogic/qede/qede_main.c	2017-04-20 18:54:40.839307597 +0200
@@ -150,8 +150,7 @@
 }
 
 #ifdef CONFIG_QED_SRIOV
-static int qede_set_vf_vlan(struct net_device *ndev, int vf, u16 vlan, u8 qos,
-			    __be16 vlan_proto)
+static int qede_set_vf_vlan(struct net_device *ndev, int vf, u16 vlan, u8 qos)
 {
 	struct qede_dev *edev = netdev_priv(ndev);
 
@@ -160,9 +159,6 @@
 		return -EINVAL;
 	}
 
-	if (vlan_proto != htons(ETH_P_8021Q))
-		return -EPROTONOSUPPORT;
-
 	DP_VERBOSE(edev, QED_MSG_IOV, "Setting Vlan 0x%04x to VF [%d]\n",
 		   vlan, vf);
 
@@ -512,7 +508,7 @@
 	.ndo_change_mtu = qede_change_mtu,
 #ifdef CONFIG_QED_SRIOV
 	.ndo_set_vf_mac = qede_set_vf_mac,
-	.extended.ndo_set_vf_vlan = qede_set_vf_vlan,
+	.ndo_set_vf_vlan = qede_set_vf_vlan,
 #endif
 	.ndo_vlan_rx_add_vid = qede_vlan_rx_add_vid,
 	.ndo_vlan_rx_kill_vid = qede_vlan_rx_kill_vid,
